Plugin/QmlApi/source/QmlArcGISArView.cpp

```diff
@@ -0,0 +1,270 @@
+    return;
+
+  m_sceneView = sceneView;
+
+  // set properties
+  m_sceneView->setProperty("spaceEffect", 1); // SpaceEffect::Transparent
+  m_sceneView->setProperty("atmosphereEffect", 0); // AtmosphereEffect::None
+  m_sceneView->setProperty("manualRendering", true);
+
+  if (m_tmcc)
+    m_sceneView->setProperty("cameraController", QVariant::fromValue(m_tmcc));
+
+  emit sceneViewChanged();
+}
+
+/*!
+  \internal
+ */
+QObject* QmlArcGISArView::transformationMatrixCameraController() const
+{
+  return m_tmcc;
+}
+
+void QmlArcGISArView::setTransformationMatrixCameraController(QObject* tmcc)
+{
+  if (!tmcc || m_tmcc || !assertClassName(tmcc, "QmlTransformationMatrixCameraController"))
+    return;
+
+  m_tmcc = tmcc;
+
+  // set TMCC to scene view
+  if (m_sceneView)
+    m_sceneView->setProperty("cameraController", QVariant::fromValue(m_tmcc));
+
+  emit transformationMatrixCameraControllerChanged();
+}
+
+/*!
+  \brief Sets the initial transformation used to offset the originCamera.
+
+  The initial transformation is based on an AR point determined via existing plane hit detection
+  from `screenPoint`. If an AR point cannot be determined, this method will return `false`.
+
+  \list
+    \li \a x - The x-coordinate of the screen point to determine the `initialTransformation` from.
+    \li \a y - The y-coordinate of the screen point to determine the `initialTransformation` from.
+  \endlist
+ */
+void QmlArcGISArView::setInitialTransformation(float x, float y)
+{
+  // Use the `hitTestInternal` method to get the matrix of `screenPoint`.
+  const std::array<double, 7> hitResult = hitTestInternal(x, y);
+  if (hitResult[3] == 0) // quaternionW shouldn't be 0
+    return;
+
+  emit initialTransformationChanged(0.0, 0.0, 0.0, 1.0, hitResult[4], hitResult[5], hitResult[6]);
+}
+
+/*!
+  \qmlmethod Point ArcGISArView::screenToLocation(Point screenPoint)
+
+  \brief Gets the location in the real world space corresponding to the screen point.
+*/
+QObject* QmlArcGISArView::screenToLocation(QObject* screenPoint) const
+{
+  if (!m_sceneView || !assertClassName(screenPoint, "QmlPoint"))
+    return nullptr;
+
+  // gets the position of the screen point
+  bool isOk = false;
+  const double x = screenPoint->property("x").toDouble(&isOk);
+  if (!isOk)
+    return nullptr;
+
+  const double y = screenPoint->property("y").toDouble(&isOk);
+  if (!isOk)
+    return nullptr;
+
+  // gets the current viewpoint camera
+  QObject* currentViewpointCamera = m_sceneView->property("currentViewpointCamera").value<QObject*>();
+  if (!currentViewpointCamera)
+    return nullptr;
+
+  // converts the point from the screen space to the 3D space
+  const std::array<double, 7> hitResult = ArcGISArViewInterface::hitTestInternal(x, y);
+
+  // calculates the matrix transformations and creates the QmlPoint location.
+  QVariant location;
+  QMetaObject::invokeMethod(m_qmlWrapper, "screenToLocation", Qt::DirectConnection,
```

Comment:
What object is `screenToLocation` being invoked on? I don't see any such method in the API that takes this many arguments.

Reply:
This can be used by the users to place an 3D object in the 3D scene. For the moment, the sample for that is not implemented, it's in the list for the next PR.